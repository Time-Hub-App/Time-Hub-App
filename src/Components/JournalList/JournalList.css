.journallist {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  gap: 20px;
  padding: 20px;
  font-weight: bold;
}

.delete {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  border: black solid 2px;
  border-radius: 10px;
}

.item {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: stretch;
  font-weight: bold;
}

.item h3 {
  margin: 10px 10px;
}

.item h3:hover {
  cursor: pointer;
  text-decoration: underline;
}
.item h2 {
  margin: 10px 10px;
  color: black;
}

.item h2:hover {
  cursor: pointer;
  text-decoration: underline;
}
