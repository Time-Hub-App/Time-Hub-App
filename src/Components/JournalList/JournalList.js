import './JournalList.css';
import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';

import { deleteJournal, fetchJournals } from '../../Services/journals';

export default function JournalList() {
  const [journals, setJournals] = useState([]);
  const [click, setClick] = useState(false);

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchJournals();
      setJournals(data);
      setClick(false);
    };
    fetchData();
  }, [click]);

  const deleteHandler = async (id) => {
    await deleteJournal(id);
    setClick(true);
  };

  return (
    <div className="journallist">
      {journals.map((journal) => (
        <div key={journal.id} className="item">
          <h3 onClick={() => deleteHandler(journal.id)}>Delete</h3>
          <Link to={`/journals/${journal.title}`}>{journal.title}</Link>
        </div>
      ))}
    </div>
  );
}
