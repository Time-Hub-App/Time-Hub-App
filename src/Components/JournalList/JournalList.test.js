import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import JournalList from './JournalList';

it('should render a journallist presentational component', () => {
  const { container } = render(
    <MemoryRouter>
      <JournalList />
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});
