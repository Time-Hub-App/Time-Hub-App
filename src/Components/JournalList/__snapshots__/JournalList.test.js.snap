// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a header 1`] = `
<div>
  <div
    class="journallist"
  />
</div>
`;

exports[`should render a journallist presentational component 1`] = `
<div>
  <div
    class="journallist"
  />
</div>
`;
